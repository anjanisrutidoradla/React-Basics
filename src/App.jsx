import React from "react";
import "./App.css";
import { AddRequestForm, RequestList } from './components/ServiceRequest'
import Nav from './components/Navbar'
import RequestChart from "./components/Chart";
import Landing from './components/Landing'
import {
  Routes,
  Route
} from "react-router-dom";
import request_list from './initial-data.json';
import { useState } from "react";


function App() {
  const [reqList, setreqList] = useState(request_list);
  const addnewRequest = (newRequest) => {
    setreqList([
      ...reqList, newRequest
    ]);
  };
  return (
    <>
      <Nav />
      <Routes>
        <Route path="/" element={<Landing />} />
        <Route path="/list" element={<RequestList data = {reqList} set_data = {setreqList}  />} />
        <Route path="/add" element={<AddRequestForm addnewRequest={addnewRequest}/>} />
        <Route path="/chart" element={<RequestChart data = {reqList} />} />
      </Routes>
    </>
  );
}

export default App;