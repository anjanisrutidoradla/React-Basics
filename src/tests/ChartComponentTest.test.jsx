import { beforeEach, describe, expect, test } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import App from "../App";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import React from "react";
import { createServiceRequest } from "./testUtils";

beforeEach(() => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
});

describe("Test Cases For Charting -> 312 App ", () => {
  test("(2 pts) Test Chart Component with initial-data.json data", async () => {
    // Go to the Visualize requests
    await userEvent.click(screen.getByText("Visualize requests"));
    // Compare with snapshot with default initial-data.json requests
    expect(
      document.getElementsByClassName("form-contain")[0].innerHTML
    ).toMatchSnapshot();
  });

  test("(3 pts)Test Chart Component After Appending Some Data to the request", async () => {
    // Create a service request
    let iname = "Test Name";
    let isdesc = "Test Short Desc";
    let ilongdesc = "Test Long Description";
    let iemailId = "email@email.com";
    await createServiceRequest(iname, isdesc, ilongdesc, iemailId);

    await userEvent.click(screen.getByText("Visualize requests"));

    // Compare with snapshot after adding one service request
    expect(
      document.getElementsByClassName("form-contain")[0].innerHTML
    ).toMatchSnapshot();
  });
});
