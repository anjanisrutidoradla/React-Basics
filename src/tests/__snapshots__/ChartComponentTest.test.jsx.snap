// Vitest Snapshot v1

exports[`Test Cases For Charting -> 312 App  > (2 pts) Test Chart Component with initial-data.json data 1`] = `"<div class=\\"simple-cloud\\"><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(4, 101, 107); font-size: 24px;\\">stop.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(5, 72, 102); font-size: 24px;\\">them.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(2, 93, 112); font-size: 24px;\\">lights</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(155, 13, 51); font-size: 24px;\\">some</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(196, 157, 1); font-size: 48px;\\">all</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(2, 135, 75); font-size: 24px;\\">block</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(191, 3, 163); font-size: 24px;\\">night.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(4, 44, 102); font-size: 24px;\\">give</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(186, 1, 124); font-size: 24px;\\">neighbors</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(13, 53, 132); font-size: 24px;\\">fix</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(7, 132, 45); font-size: 24px;\\">on</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(4, 10, 94); font-size: 48px;\\">Please</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(201, 14, 198); font-size: 24px;\\">jackhammering</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(221, 110, 6); font-size: 24px;\\">make</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(17, 170, 24); font-size: 24px;\\">trash</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(202, 18, 226); font-size: 24px;\\">and</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(216, 161, 8); font-size: 24px;\\">out.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(62, 1, 132); font-size: 24px;\\">my</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(204, 74, 14); font-size: 36px;\\">are</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(73, 142, 8); font-size: 24px;\\">day</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(198, 119, 15); font-size: 24px;\\">The</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(51, 12, 142); font-size: 24px;\\">them</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(110, 127, 12); font-size: 24px;\\">cans.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(11, 124, 102); font-size: 24px;\\">My</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(165, 13, 64); font-size: 24px;\\">me</span></div>  "`;

exports[`Test Cases For Charting -> 312 App  > (3 pts)Test Chart Component After Appending Some Data to the request 1`] = `"<div class=\\"simple-cloud\\"><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(216, 161, 8); font-size: 24px;\\">out.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(11, 124, 102); font-size: 24px;\\">My</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(13, 53, 132); font-size: 24px;\\">fix</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(4, 101, 107); font-size: 24px;\\">stop.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(7, 132, 45); font-size: 24px;\\">on</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(155, 13, 51); font-size: 24px;\\">some</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(2, 93, 112); font-size: 24px;\\">lights</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(204, 74, 14); font-size: 36px;\\">are</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(110, 127, 12); font-size: 24px;\\">cans.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(0, 2, 96); font-size: 24px;\\">Description</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(165, 13, 64); font-size: 24px;\\">me</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(202, 18, 226); font-size: 24px;\\">and</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(4, 44, 102); font-size: 24px;\\">give</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(62, 1, 132); font-size: 24px;\\">my</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(5, 72, 102); font-size: 24px;\\">them.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(191, 3, 163); font-size: 24px;\\">night.</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(13, 118, 135); font-size: 24px;\\">Long</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(221, 211, 13); font-size: 24px;\\">Test</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(51, 12, 142); font-size: 24px;\\">them</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(4, 10, 94); font-size: 48px;\\">Please</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(2, 135, 75); font-size: 24px;\\">block</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(196, 157, 1); font-size: 48px;\\">all</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(221, 110, 6); font-size: 24px;\\">make</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(198, 119, 15); font-size: 24px;\\">The</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(73, 142, 8); font-size: 24px;\\">day</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(186, 1, 124); font-size: 24px;\\">neighbors</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(201, 14, 198); font-size: 24px;\\">jackhammering</span><span class=\\"tag-cloud-tag\\" style=\\"margin: 0px 3px; vertical-align: middle; display: inline-block; color: rgb(17, 170, 24); font-size: 24px;\\">trash</span></div>  "`;
